core/src/main/java/org/mskcc/cbio/portal/dao/DaoMutation.java

```diff
@@ -635,17 +635,19 @@ public static boolean hasAlleleFrequencyData (int geneticProfileId, int sampleId
             pstmt.setString(6, event.getProteinChange());
             pstmt.setString(7, event.getMutationType());
             rs = pstmt.executeQuery();
-            if (rs.next()) {
-                return extractMutationEvent(rs);
-            }
-            else {
-                return null;
+            while(rs.next()) {
```

Comment:
why this while? Is mysql query not returning an exact match?

Reply:
@pieterlukasse this is _just in case_ there is more than one result returned. The query looking for a match against `TUMOR_SEQ_ALLELE`, and I didn't run into any such cases even when purposefully introducing variants that overlapped the first 255 nuc's, but this is just a safety put in place I guess.

Reply:
but the previous code returns the first match, while this one does as well. The only difference is that the new code is slightly more confusing. Can you please revert this one?

Reply:
@pieterlukasse sure, np 👍

Reply:
Perhaps this loop is to handle a subtlety that we encountered once or twice in the past. We saw cases where the TUMOR_SEQ_ALLELE string contained a non-ascii encoded character (I cannot remember whether it was unicode or utf8). There was a difference in function output between the java string comparison library and the mysql `=` operator for the where clause. If I remember correctly, the mysql `=` operator is more tolerant of difference across encodings. I cannot remember what the exact character was but it may have been that a hyphen (ASCII code 0x2D) evaluates to "equal" to alternative unicode horizontal stroke characters. One example I found now is comparisons between accented letters such ä (LATIN SMALL LETTER A WITH DIAERESIS) and unaccented ascii coded letters. In mysql the query `select 'a' = _utf8 x'c3a4';` returns 1 (true).  (_utf8 x'c3a4' is the utf representation of ä)

Because we are dropping the unique index, it is now hypothetically possible that two mutation_event records could exist with the only difference being that one has TUMOR_SEQ_ALLELE value 'tta' and the other has TUMOR_SEQ_ALLELE value 'ttä' and the select query might return both values. I know this is a contrived example but we did see one or two cases like this.
So the loop here would make sure that the correct string was linked to if somehow this situation occurred.